blocks/lists.ts

```diff
@@ -352,15 +352,14 @@ blocks['lists_getIndex'] = {
     const modeMenu = fieldRegistry.fromJson({
       type: 'field_dropdown',
       options: MODE,
-    });
+    }) as FieldDropdown;
     modeMenu.setValidator(
-        /**
-         * @param {*} value The input value.
-         * @this {FieldDropdown}
-         */
-        function(value) {
+        /** @param value The input value. */
+        function(this: FieldDropdown, value: string) {
           const isStatement = (value === 'REMOVE');
-          this.getSourceBlock().updateStatement_(isStatement);
+          (this.getSourceBlock() as GetIndexBlock)
+              .updateStatement_(isStatement);
+          return undefined;
```

Comment:
Why do you need to explicitly return undefined?

[Edit: Ah I see there are many places in this file that do this, so I assume it's just for consistency / clarity?]

Reply:
Because `.setValidator` does not accept `void` functions.  And I could add `: undefined` to the validator function expression, but then it complains that there is no `return` statement.  So I just added the return statement and let it infer the return type.

Reply:
Ah ok, can you file a bug for fixing `setValidator` to accept `void` functions? Because it definitely should. If no value (or a value of `undefined`) is returned from a validator, we consider the original value valid.

Reply:
So, I wondered about that, but apparently in TypeScript a function that has its return type declared as `void` does not mean "no return statement" or "no return statement, or only returns `undefined`", but [instead means "caller promises to ignore any returned value"](https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-returning-non-void-assignable-to-function-returning-void) which is obviously not true of validators.

If there is a way to tell TypeScript to infer that a function with no `return` statement is actually `(/*...*/) => undefined` rather than a `=> void` then that would be great, but I'm not aware of such.

Reply:
but but but... a function which has no return statement /is/ actually `(/* ... */) => undefined` lol

I guess this is why migrations are hard :P

Reply:
I know.  I guess the problem is that `tsc` has to decide the return type of a `return`-less function.  Is it `void` or is it `undefined`?  It would be nice if it could be "OK, I'll treat it as `: void&undefined` (i.e., satisifies either) but there's probably some reason that's a bad idea.  Alternatively it could allow you to declare it as `: undefined` without actually having to have the `return` statement.  But alas it won't allow that either.   ¯\\\_(ツ)\_/¯